### ISSUE TITLE
Sequence of steps in Inspector gets MNU and turns into random other object (3.0.93-4)

### ISSUE THREAD

**Issue by LisaAlmarode** [ericwinger/Jade#716]
I have a sequence of operations that ends up with a corrupt inspector. Looks like something in the inspector is getting GCed. It's quite reproducible for my application, but it does seem to require something specific, I've got it isolated to a simple example case.

The sequence is: open inspector on collection, inspect one item, execute code to open an instance of a helper class referencing that item, execute code in the inspector on the helper... this works reliably one time, but after that executing the same code in the same inspector gets MNU and the inspector starts show other objects. 

These are the three classes to reproduce the simple case:

```
Class {
        #name : 'MyListItem',
        #superclass : 'Object',
        #instVars : [
                'aa',
                'bb'
        ],
        #category : 'RowanSample1-Core'
}

{ #category : 'Instance creation' }
MyListItem class >> create [
        | n |
        n := self new.
        n aa: 'aaa ' copy.
        n bb: 'bbb ' copy.
        ^n
]

{ #category : 'accessing' }
MyListItem >> aa [
        ^aa
]

{ #category : 'accessing' }
MyListItem >> aa: object [
        aa := object
]

{ #category : 'accessing' }
MyListItem >> bb [
        ^bb
]

{ #category : 'accessing' }
MyListItem >> bb: object [
        bb := object
]
```

```
Class {
        #name : 'MyOrderedCollection',
        #superclass : 'OrderedCollection',
        #instVars : [
                'itemDetails'
        ],
        #category : 'RowanSample1-Core'
}

{ #category : 'instance creation' }
MyOrderedCollection class >> create [
        | n |
        n := self new.
        10 timesRepeat: [n add: MyListItem create].
        n itemDetails: Array new.
        10 timesRepeat: [n itemDetails add: MyListItem create].
        ^n
]

{ #category : 'accessing' }
MyOrderedCollection >> itemDetails [
        ^itemDetails
]

{ #category : 'accessing' }
MyOrderedCollection >> itemDetails: object [
        itemDetails := object
]
```

```
Class {
        #name : 'MyHelperClass',
        #superclass : 'Object',
        #instVars : [
                'myListItem'
        ],
        #category : 'RowanSample1-Core'
}

{ #category : 'instance creation' }
MyHelperClass class >> for: aListItem [
        | n |
        n := self new.
        n myListItem: aListItem.
        ^n
]

{ #category : 'behavior' }
MyHelperClass >> interestingComputation [
        ^self myListItem aa, self myListItem bb
]

{ #category : 'accessing' }
MyHelperClass >> myListItem [
        ^myListItem
]

{ #category : 'accessing' }
MyHelperClass >> myListItem: object [
        myListItem := object
]
```

Once you have these classes in the image, this is how I can reproduce it reliably:

1. inspect ```MyOrderedCollection create```

2.  In the inspector, select an item, say number 6, and inspect that.  This opens an inspector on that MyListItem.

3.  In the text pane for the MyListItem inspector, enter ```MyHelperClass for: self```, 

![image](https://user-images.githubusercontent.com/13142166/74314272-010fcf80-4d2a-11ea-816d-4e7893e938dd.png)

and inspect, which opens an inspectors on a MyHelperClass.

4.  In the text pane for this inspector, enter ```self interestingComputation```
and display.  So far so good.

![image](https://user-images.githubusercontent.com/13142166/74315513-72508200-4d2c-11ea-93f0-be0c2c172a55.png)

Then try this again... this time an inspect it on the results. Sometimes it takes a third inspect or display.  Now, it can't do it:

![image](https://user-images.githubusercontent.com/13142166/74314404-3ae0d600-4d2a-11ea-9875-e448f6576c83.png)

Close this.  Go back to the inspector on MyHelperClass.  select the instance variable myListItem (say no to keep unsaved changes). Double click on it.  Looks fine.  Use the back arrow to return to the MyHelperClass instance.

But it's transformed into something it shouldn't be!

![image](https://user-images.githubusercontent.com/13142166/74314561-754a7300-4d2a-11ea-88b6-9471e687bb2f.png)

**Comment by ericwinger** [comment]
Fix is a single method rowan services fix. https://github.com/GemTalk/Rowan/compare/1d22cecb16aa...e06e88806fbe

**Comment by ericwinger** [comment]
Fixed in 3.0.93pre-release5. It was a garbage collection problem. Was not retaining the root object of the original inspector.
